### PR TITLE
Add terminationGracePeriodDuration to proxy config

### PR DESCRIPTION
**Pull request by liamawhite**
This field will be consumed by https://github.com/istio/istio/pull/11298.

Part of the work for https://github.com/istio/istio/issues/7136.

### CODE REVIEW

File: mesh/v1alpha1/proxy.proto [hunk 1]
```diff
@@ -164,4 +164,11 @@ message ProxyConfig {
 
   // Tracing configuration to be used by the proxy.
   Tracing tracing = 19;
+
+  // The time in seconds that Istio will allow Envoy to complete requests before terminating it.
+  // This currently supports all HTTP based protocols (HTTP, HTTP2, GRPC, etc.) but not HTTPS.
+  // On receiving SIGTERM or SIGINT Istio instructs Envoy to respond with 503 to any new inbound requests
+  // It will continue to allow new outbound and any already open connections. Once the duration is reached, 
+  // Istio kills the Envoy process. MUST be >=1s (e.g., _1s/1m/1h_)
+  google.protobuf.Duration termination_grace_period_duration = 20;
```
[the remark this answers is not in the document]
Comment (louiscryan) [reply mesh/v1alpha1/proxy.proto]: @liamawhite - can you open a tracking issue targeted for the nebulous future to allow this to be configured per-deployment or something more refined than global.

File: mesh/v1alpha1/proxy.proto [hunk 2]
```diff
@@ -164,4 +164,11 @@ message ProxyConfig {
 
   // Tracing configuration to be used by the proxy.
   Tracing tracing = 19;
+
+  // The time in seconds that Istio will allow Envoy to complete requests before terminating it.
+  // This currently supports all HTTP based protocols (HTTP, HTTP2, GRPC, etc.) but not HTTPS.
+  // On receiving SIGTERM or SIGINT Istio instructs Envoy to respond with 503 to any new inbound requests
```
[the remark this answers is not in the document]
Comment (louiscryan) [reply mesh/v1alpha1/proxy.proto]: @liamawhite  can you link to the Envoy documentation associated with this behavior as an example implementation

[the remark this answers is not in the document]
Comment (liamawhite) [reply mesh/v1alpha1/proxy.proto]: See istio/istio#11298, but this is a no go due to the tracing limitations.

File: mesh/v1alpha1/proxy.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (liamawhite) [reply mesh/v1alpha1/proxy.proto]: We will be implementing graceful shutdown via a different mechanism now, one that uses the envoy drain-time and parent shutdown times instead.

**Pull request closed without merging**